### DISCUSSION TITLE
Getting NULL with swell.cart.getShippingRates()

### DISCUSSION THREAD

**Discussion by ItsAndrewM** [swellstores/community#417] [Help] [answered]
Hi there,

I've been running into an issue trying to get shipping rates in a custom checkout.  As another user had added, and according to the docs, a country ISO code (ive used "CA") needs to be added to the shipping of a cart.  I've done so but I'm still getting null from `swell.cart.getShippingRates() ` despite that.

Any help would be appreciated.

Thank you.

**Comment by ItsAndrewM** [comment]
I should add that I've attempted to do what another post mentioned which was to change the base currency to a different currency then change it back and have still received the same error.  Shipping method page doesnt appear to even allow the user to navigate to the /checkout/[id]/shipping-method unless directly.  And this is the shipping method panel: This checkout id does have a shipping address attached (as well as a country code - "CA").  I've also inlcuded what my shipping rates look like. 
![image](https://github.com/swellstores/community/assets/112970534/4e86a566-89fa-432b-a092-451c19d59f7c)

![image](https://github.com/swellstores/community/assets/112970534/795e55ee-d48d-40a3-a0ea-f91cea7ff76c)

![image](https://github.com/swellstores/community/assets/112970534/7bd50bea-433d-45d3-83b4-79ff7af67067)

**Reply by coderreco** [reply]
I would recommend going into the cart that is attached to your session in your Swell admin panel. Click 'View in Console' from the actions menu and check your shipping object. I will get a shipping rating off of just this:
```
  "shipping": {
    "country": "CA",
  },
```
Then I would suggest going back to the cart in your admin panel, and updating the shipping address in the panel on the right hand side. 
Or in the console you could run a PUT on that cart with the same code as above:
```
  "shipping": {
    "country": "CA",
  },
```
Then click 'Edit Shipping' under the totals in the cart UI and see if there is a button that says 'Get Rates'. If there is click that, then check for the rating in the console. If it is there, go back to the checkout page and see if you have shipping options?

**Reply by ItsAndrewM** [reply, marked as answer]
I tried your answer and didn't have any luck.  However, I did figure out what the issue was:

I noticed that around 5 of our 522 products actually did include the shipping panel on the product UI.
![image](https://github.com/swellstores/community/assets/112970534/cdc64acf-e333-4f60-8d54-ab860ac428b2)

I exported that product and compared it against a product that didn't have the shipping panel: 
![image](https://github.com/swellstores/community/assets/112970534/44dfc9c9-2b58-43d1-854e-6b63204fcae0)

The only thing missing between the two products (outside of variations/options) was that the "delivery" column didnt include the value "shipment": 
![image](https://github.com/swellstores/community/assets/112970534/1d11ad30-524e-4c46-9d72-814eb85219d5)

After updating that value the shipping panel appeared in that products UI.

I added the product to the cart and then went through the checkout and sure enough the shipping rates appeared with `swell.cart.getShippingRates():`
![image](https://github.com/swellstores/community/assets/112970534/9c1b71fd-66f5-4167-ab9f-f2f869e0a308)
